lib/core/src/lib/progress/progress.component.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * @license
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Input, ViewEncapsulation } from '@angular/core';
+import { MatProgressBarModule, ProgressBarMode } from '@angular/material/progress-bar';
+import { MatProgressSpinnerModule, ProgressSpinnerMode } from '@angular/material/progress-spinner';
+import { CommonModule } from '@angular/common';
+import { ThemePalette } from '@angular/material/core';
+import { MatButtonModule } from '@angular/material/button';
+
+export type ProgressVariant = 'bar' | 'spinner' | undefined;
+export type ProgressMode = ProgressBarMode | ProgressSpinnerMode;
+export type ProgressColor = ThemePalette;
+
+@Component({
+    selector: 'adf-progress',
+    standalone: true,
+    imports: [CommonModule, MatProgressBarModule, MatProgressSpinnerModule, MatButtonModule],
```

Comment:
Is the import for MatButton needed?

Reply:
good finding, removed the import, thanks